x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/Autoscaling.java

```diff
@@ -156,9 +181,15 @@ boolean isSnapshot() {
             )
         );
     }
-
-    protected XPackLicenseState getLicenseState() {
-        return XPackPlugin.getSharedLicenseState();
+    @Override
+    public void extensionPlugin(Plugin plugin) {
```

Comment:
`Autoscaling` receives information on an extending plugin and keeps a reference to it.